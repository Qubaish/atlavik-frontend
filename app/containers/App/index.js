/**
 *
 * App.js
 *
 * This component is the skeleton around the actual pages, and should only
 * contain code that should be seen on all pages. (e.g. navigation bar)
 *
 */

import React from 'react';
import { Helmet } from 'react-helmet';
import { Switch, Route } from 'react-router-dom';
import styled from 'styled-components';

import HomePage from 'containers/HomePage/Loadable';
import NotFoundPage from 'containers/NotFoundPage/Loadable';

import Header from 'components/Header';
import Footer from 'components/Footer';

import GlobalStyle from '../../global-styles';

const AppWrapper = styled.div`
  margin: 0 auto;
  display: flex;
  min-height: 100%;
  flex-direction: column;
`;

export default function App() {
  return (
    <AppWrapper>
      <Helmet titleTemplate="Atlavik App" defaultTitle="Atlavik">
        <meta name="description" content="Hotels list" />
      </Helmet>
      <Header />
      <Switch>
        <Route exact path="/" component={HomePage} />
        <Route component={NotFoundPage} />
      </Switch>
      <Footer />
      <GlobalStyle />
    </AppWrapper>
  );
}
