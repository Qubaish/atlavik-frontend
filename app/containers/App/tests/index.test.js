import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import App from '../index';

describe('<App />', () => {
  it('should render and match the snapshot', () => {
    const container = shallow(
      <App />
  );
  expect(shallowToJson(container)).toMatchSnapshot();
  });
});
