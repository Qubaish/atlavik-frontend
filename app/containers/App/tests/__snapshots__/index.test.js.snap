// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should render and match the snapshot 1`] = `
<App__AppWrapper>
  <HelmetWrapper
    defaultTitle="Coding test"
    defer={true}
    encodeSpecialCharacters={true}
    titleTemplate="Atlavik App"
  >
    <meta
      content="Hotels list"
      name="description"
    />
  </HelmetWrapper>
  <Header />
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
    />
  </Switch>
  <Footer />
  <GlobalStyleComponent />
</App__AppWrapper>
`;
